src/yoroi-wallets/cardano/transactionManager.test.ts

```diff
@@ -0,0 +1,148 @@
+      }),
+    )
+    expect(txManager.getPerRewardAddressCertificates()).toEqual({
+      e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae: {
+        '0a8962dde362eef1f840defe6f916fdf9701ad53c7cb5dd4a74ab85df8e9bffc': {
+          submittedAt: '2021-09-13T18:42:10.000Z',
+          epoch: 156,
+          certificates: [
+            {
+              kind: 'StakeDeregistration',
+              rewardAddress: 'e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae',
+            },
+          ],
+        },
+      },
+    })
+
+    // tx cache mutation
+    await txManager.doSync(mockedAddressesByChunks)
+
+    expect(txManager.getTransactions()).toEqual({
+      [mockTx.id]: mockTx,
+      ...fromPairs(mockedHistoryResponse.transactions.map((t) => [t.hash, toCachedTx(t)])),
+    })
+    expect(txManager.getPerAddressTxs()).toEqual(
+      perAddressTxsSelector({
+        transactions: {
+          [mockTx.id]: mockTx,
+          ...fromPairs(mockedHistoryResponse.transactions.map((t) => [t.hash, toCachedTx(t)])),
+        },
+        perAddressSyncMetadata: {},
+        bestBlockNum: null,
+      }),
+    )
+    expect(txManager.getPerRewardAddressCertificates()).toEqual({
+      e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae: {
+        '0a8962dde362eef1f840defe6f916fdf9701ad53c7cb5dd4a74ab85df8e9bffc': {
+          submittedAt: '2021-09-13T18:42:10.000Z',
+          epoch: 156,
+          certificates: [
+            {
+              kind: 'StakeDeregistration',
+              rewardAddress: 'e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae',
+            },
+          ],
+        },
+        '54ab3dc8e717040b9b4c523d0756cfc59a30f107e053b4cd474e11e818be0ddg': {
+          submittedAt: '2022-06-12T23:46:47.000Z',
+          epoch: 210,
+          certificates: [
+            {
+              kind: 'StakeRegistration',
+              rewardAddress: 'e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae',
+            },
+            {
+              kind: 'StakeDelegation',
+              poolKeyHash: '8a77ce4ffc0c690419675aa5396df9a38c9cd20e36483d2d2465ce86',
+              rewardAddress: 'e0acab7e493ece4c1e6ae627ef9f5f7c9b1063e599e4aa91f87f0d58ae',
+            },
+          ],
+        },
+      },
+    })
+  })
+})
+
+describe('memos manager', () => {
+  it('works', async () => {
+    const storage = rootStorage.join('memos/')
+    const memosManager = await makeMemosManager(storage)
+
+    expect(memosManager.getMemos()).toEqual({})
+
+    await memosManager.saveMemo('fake-tx-id-1', 'Send money to my friend')
+    await memosManager.saveMemo('fake-tx-id-2', 'Send money to my girlfriend')
+
+    expect(memosManager.getMemos()).toEqual({
+      'fake-tx-id-1': 'Send money to my friend',
+      'fake-tx-id-2': 'Send money to my girlfriend',
+    })
+
+    await memosManager.clear()
+
+    expect(memosManager.getMemos()).toEqual({})
+  })
+})
+
+// mocks
+
+const mockStorage: YoroiStorage = {
```

Comment:
can you create a different storage for each test, and seed it with whatever data it needs?
something like:
```
storage.join("txs/").setItem("txids", [mockTx.id])
storage.join("txs/").multiSet([[txids[0], mockTx])
```

Reply:
can you do this?
```
const txStorage = storage.join("txs/")
txStorage.setItem("txids", [mockTx.id])
txStorage.multiSet([[txids[0], mockTx])
// ...whatever setup....

const transactionManager = makeTransactionManager(txStorage)
```

Reply:
I'm not getting to model this to work.  I need to test it further anyway.

Reply:
@wolverineks tested. And its not possible its implementation because its creating a path like: `.../join/join`

Let me know if you have another idea to implement this.